Create issue templates and instructions for release testing

jashapiro opened the pull request:
I used github's features to create some issue templates.

Originally, I was just going to create the create an issue template for new releases, but it seemed easy enough to just make all the templates at once. 

I used the provided templates for features and bugs, with some modifications, especially for the bug report one.
I expect features to be the one we use most of the time (or no template), and it is the most general.

The release checklist is the one that should need the most review though:
- Did I describe the steps to test a release candidate sufficiently? 
- Should I include more detail about what happens after all the tests pass? (Instructions for actually creating a release?)

jaclyn-taroni commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,27 @@
+---
+name: Bug report
```
This looks good 👍🏻

jaclyn-taroni commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+---
+name: Feature request
```
This one looks good, too 👍🏻

jaclyn-taroni commented on .github/ISSUE_TEMPLATE/release-checklist.md:
```diff
@@ -0,0 +1,20 @@
+
+## Steps for a new release of `scpcaTools`
+
+- [ ] Are all dependencies/issues expected for this release are resolved? Add any that are not resolved as dependencies for this issue.
```
Do you mean mark this issue (the release) as blocked by the unresolved issues in ZenHub?

jashapiro replied:
Yes, I will reword.

jaclyn-taroni commented on .github/ISSUE_TEMPLATE/release-checklist.md:
````diff
@@ -0,0 +1,20 @@
+  - [ ] If necessary, update the `scpca-nf` workflow to accommodate changes in the `scpcaTools` package  
+  - [ ] Perform a test run of the workflow with:
+```nextflow run AlexsLemonade/scpca-nf -r "<TESTBRANCH>" -profile batch```
+You may want to specify libraries or runs for testing.
````
I believe your point here is: You might actually want to run something on some data. Is that correct?

jashapiro replied:
Sort of... There are default libraries in the workflow, but if there is a particular system to test, others might be good choices. I will reword.

jaclyn-taroni reviewed (commented):
This looks pretty good! I had a couple questions about the release checklist, which I believe can be resolved with rewording things. 

Regarding this question:

> Should I include more detail about what happens after all the tests pass? (Instructions for actually creating a release?)

My inclination is yes. I don't expect that it will need to be lengthy. I think you could change the current section title to make it more explicitly about testing/preparing for a release and add a new section `Creating a release`

jashapiro commented:
I updated some wording and added release instructions. The package was failing tests yesterday because it seems to be failing to download and install some dependencies. Not fully sure what to do about that, and it usually seems to be transient. If it fails again I might look into it a bit more in the interests of avoiding similar problems in the future.

jaclyn-taroni commented on .github/ISSUE_TEMPLATE/release-checklist.md:
```diff
@@ -0,0 +1,29 @@
+### Preparing for the release and testing
+
+- [ ] Are all of the issues planned for this release resolved? If there are any issues that are unresolved, mark this issue as blocked by those on ZenHub.
+- [ ] Did all automated tests and the Docker build in this repo pass in CI? (You may have to wait for the docker image to build. Check [Github actions](https://github.com/AlexsLemonade/scpcaTools/actions/workflows/build-docker.yaml) before proceeding to the next step
```
```suggestion
- [ ] Did all automated tests and the Docker build in this repo pass in CI? (You may have to wait for the docker image to build. Check [Github actions](https://github.com/AlexsLemonade/scpcaTools/actions/workflows/build-docker.yaml) before proceeding to the next step)
```

Or maybe the `)` goes after "wait for the docker image to build."

jashapiro replied:
Just made it not a parenthetical.

jaclyn-taroni reviewed (approved):
Caught one missing ), but other than that LGTM!

jashapiro commented:
Now that the tests are passing, I will merge this.
